### ISSUE TITLE
Amplify CI/CD - Branch changes are sometimes not detected

### ISSUE THREAD

**Issue by Jordan-Eckowitz** [aws-amplify/amplify-console#1154]
**Describe the bug**
Sometimes when I merge changes into a branch Amplify does not detect the update on Github. It's happened on multiple branches I am trying to deploy. I have tried `Reconnect Repository` as well as disabling then re-enabling branch `Auto-build` but it doesn't do anything. Most of the time my builds work so I know my setup works but this has started to occur with more regularity. The only solution I've found to fix this is deleting the branch and re-adding it when this occurs. 

**Expected behavior**
When I merge a tracked branch Amplify will deploy an update to the statically hosted site. 

**Screenshots**

![image](https://user-images.githubusercontent.com/19767538/96731695-89e50d80-1385-11eb-9909-4266d14beadf.png)

**Comment by behrooziAWS** [comment]
Can you send a send app id/region to aws-amplify-customer@amazon.com and reference this issue?

**Comment by Jordan-Eckowitz** [comment]
@behrooziAWS I sent the email but still waiting on a reply...

**Comment by Jordan-Eckowitz** [comment]
@behrooziAWS I got an email 2 weeks ago from someone saying they're looking into this. I have followed up multiple times but they never reply.

Can I please get an update. This has become a significant issue for my team.

**Comment by nimacks** [comment]
@Jordan-Eckowitz , I'm looking into this. Does your team run more than 5 concurrent builds at a time?

**Comment by hsspain** [comment]
@Jordan-Eckowitz We have made improvements to our systems that should increase the reliability of detecting branch changes.  This should no longer be an issue for you.

I am closing this ticket, but please reach out if you have further issues.

**Comment by hsspain** [comment]
Re-opening this ticket as the fix has not been fully deployed yet.  I will update this as we verify the fix is complete.

**Comment by Jordan-Eckowitz** [comment]
> Re-opening this ticket as the fix has not been fully deployed yet. I will update this as we verify the fix is complete.

Yeah, just saw its still not working

**Comment by Jordan-Eckowitz** [comment]
Hi @hsspain - has this fix been deployed yet?

**Comment by adambschroeder** [comment]
Also experiencing this when pushing changes to a CodeCommit branch. Change not detected in Amplify thus the build never runs.

**Comment by nimacks** [comment]
We have a fix for this issue which will begin rolling out in a week. Stay tuned. I'll update this thread next week.

**Comment by nimacks** [comment]
We have started rolling out a fix for this issue which should be available to all regions next week.

**Comment by adambschroeder** [comment]
Hi @nimacks , curious if we should expect to see this change yet? Thank you for looking into.

**Comment by abhi7cr** [comment]
Hey @adambschroeder ,  we pushed a fix to address codecommit pushes not triggering builds. However, there is an edge case scenario _when multiple branches are pushed at the same time, and some of them don't exist_, it short-circuits instead of skipping. We will be deploying this fix shortly, but the general happy path should work. Let us know if you face any issues.

**Comment by nimacks** [comment]
hey @Jordan-Eckowitz, can you confirm that your issue has been fixed?

**Comment by adambschroeder** [comment]
I am not seeing the fix for this yet. Will I need to delete a branch connection and redo it?

Our repository is on _CodeCommit_ and we have Amplify connected to our [develop] branch. Whenever we merge into or commit to [develop], Amplify does NOT start the build.

**Comment by Jordan-Eckowitz** [comment]
> hey @Jordan-Eckowitz, can you confirm that your issue has been fixed?

Hey @nimacks - I haven't had any issues with it this week. I'll post here if I run into any issues but so far it seems fine.

**Comment by Jordan-Eckowitz** [comment]
> I am not seeing the fix for this yet. Will I need to delete a branch connection and redo it?
> 
> Our repository is on _CodeCommit_ and we have Amplify connected to our [develop] branch. Whenever we merge into or commit to [develop], Amplify does NOT start the build.

@adambschroeder as a workaround (recommended via email by @nimacks a while ago) I have branch auto-detection enabled (`App Settings -> General`). For the branch matching patterns I've added the names of my deployed branches. It seemed to help.

@nimacks do I still need to keep that enabled or can I remove it?

**Comment by adambschroeder** [comment]
Thank you @Jordan-Eckowitz for that! I will test that out.

Without that set, the branch auto-detection still does not seem to be firing - we currently have a webhook setup to manually deploy that we have run with every update.

**Comment by behrooziAWS** [comment]
@adambschroeder The fix may have still been rolling out depending on the region you are in for CodeCommit.  It completed on March 4.
@Jordan-Eckowitz You are welcome to try removing auto-detection.  If you encounter another case where a commit does not trigger a build for a connected branch, send your commit id to aws-amplify-customer@amazon.com and turn back on auto-detection. That will enable us to review the exact webhooks we received from GitHub for that commit and identify what caused it to be ignored.

**Comment by OK-Apps** [comment]
is this supposed to be fixed in ap-southeast-2 (Sydney) region yet?

still experiencing this problem right now
when creating "branches" in CICD deployments in Amplify Console - they work initially and sometimes 2nd time around (when pushing a change to CodeCommit, but the stop triggering...

also apparently there is no way to manually "kick" a re-deployment, is it ?
if there is please tell how!

thank you!

**Comment by behrooziAWS** [comment]
@OK-Apps It has been deployed to ap-southeast-2.  If you can provide the code commit hash, we can investigate further.  If there is no job associated with the commit, there is currently no way of re-triggering a deployment in the Amplify console.

**Comment by Jordan-Eckowitz** [comment]
Hi @behrooziAWS - just ran into the same issue again. Branch change was not detected (auto-detection was still enabled and still didn't catch it...)

**Comment by vineethbhargavan** [comment]
same here!! Been happening for the past few days (maybe a few weeks!). I have tried to disconnect the branch and then connect it again. The build is triggered in this case , and we get to see the provision-> build-> deploy-> verify , all green.  So in this state - any new change on the branch is detected! ..and it proceeds with the build , however stops at - Deployment successfully completed - 100%  indefinitely!  Any new code changes (on code commit) is not detected and amplify doesn't initiate build!!. Appreciate if the amplify team could look into this!

**Comment by behrooziAWS** [comment]
@Jordan-Eckowitz can you send the commit hash to aws-amplify-customer@amazon.com for that change?
@vineethbhargavan can you send your appid, region and the commit hash of a change that did not build to aws-amplify-customer@amazon.com ?

**Comment by vineethbhargavan** [comment]
Done - @behrooziAWS

**Comment by behrooziAWS** [comment]
@vineethbhargavan Thanks! We've identified and patched an issue with CodeCommit and buildspecs that have an applications section like yours failing to trigger builds.  We'll keep you posted as the fix rolls out.

**Comment by vineethbhargavan** [comment]
Thanks for the update. Looking forward to the roll out.

**Comment by vineethbhargavan** [comment]
Any update on this? Can I get a tentative date as to when the fix will get
rolled out?

Thanks
Vineeth.

**Comment by behrooziAWS** [comment]
@vineethbhargavan the fix for your issue is now in ap-southeast-2.  It should be in all regions by EOD today.

**Comment by mikelhamer** [comment]
This fixed the same issue for me with a CodeCommit repo. @Jordan-Eckowitz my workaround to trigger builds manually when it got stuck was to use a webhook to trigger the build. 

On a side note, I've noticed that if another push is made while a build is in progress, it does not trigger a subsequent build to detect the changes. So if a developer pushes a change during a 3-5 minute window it takes for the build to complete, it will never be deployed until the next commit.

**Comment by HarborneD** [comment]
I am also experiencing this issue (in eu-west-1)

Also should this be untagged as pending-customer-response as it's still an issue for multiple customers and we're waiting on a fix?

**Comment by wewebdeveloper** [comment]
I am also experiencing this issue with code commit repo  (in ap-southeast-2)

**Comment by zaca** [comment]
After three months been really happy using amplify it stop recognizing branch changes. Now, I have to manually build my app. It's very disappointing. Because now I can't trust on this CD mechanism,  I'm thinking about going back to my python script, cloudfront and S3.

**Comment by ViniBjazevich** [comment]
I am running into this same issue. I tried reconnecting the repository and sometimes that helps but it seems to be a temporary fix. I never ran into issues with this until today and I've used amplify CI/CD consistently for months now.

**Comment by Narrator** [comment]
@ViniBjazevich Could you please share your App ID/Region?

**Comment by ViniBjazevich** [comment]
d2n3kwnhtn68zj - US-West-1

**Comment by ace-d** [comment]
I have the same issue starting from today.

**Comment by hloriana** [comment]
@ace-d please share your App ID and region

**Comment by ace-d** [comment]
Hi,
Thanks for the reply! The id is **d3b8ar5crt5pri** and region is **us-west-1**. The issue happened around 6:00 pm - 8:00 pm PST on 21st Mar.

The CI/CD feature seems working fine this morning.

**Comment by hloriana** [comment]
@ace-d @ViniBjazevich we've been working on resolving this issue. Can you confirm if this behavior is still happening?

**Comment by scottbw** [comment]
Just ran into this issue today on eu-west-2. Merge from dev -> prod doesn't trigger a build on prod

**Comment by nimacks** [comment]
@scottbw  Can you create a new issue and provide more details?
